cmd/gen-go-sample/main.go

```diff
@@ -303,28 +303,15 @@ func (g *generator) genSample(ifaceName, methName, regTag string, valSet SampleV
 		}
 	}
 
-	// TODO(pongad): handle non-unary
-	p("  resp, err := c.%s(ctx, req)", methName)
-	p("  if err != nil {")
-	p("    return err")
-	p("  }")
-	p("")
-
-	{
-		st := newSymTab(nil)
-		st.universe["$resp"] = true
-		st.scope["$resp"] = initType{desc: g.descInfo.Type[meth.GetOutputType()]}
-
-		for _, out := range valSet.OnSuccess {
-			if err := writeOutputSpec(out, st, g); err != nil {
-				return errors.E(err, "cannot write output handling code")
-			}
-		}
-
-		if len(valSet.OnSuccess) == 0 {
-			p("fmt.Println(resp)")
-			g.imports[pbinfo.ImportSpec{Path: "fmt"}] = true
-		}
+	if pf, err2 := pagingField(g.descInfo, meth); err2 != nil {
+		err = err2
+	} else if pf != nil {
+		err = g.paging(meth, pf, valSet)
```

Comment:
Not sure if in Go there are other types of calling forms besides unary and paging. If so, can we add a default option in the if-else statements here for calling forms that are not currently supported and probably just error out for now?

Reply:
Done